### PR TITLE
API docs for v2 and changes to hoverctl parts of docs

### PR DESCRIPTION
**Pull request by benjih**

### CODE REVIEW

File: docs/reference/api.md [hunk 1]
```diff
@@ -1,3 +1,87 @@
...
+
+```
+{
+    destination: "new-destination"
+}
+```
+
+
+## GET /api/v2/hoverfly/middleware
+Gets the current middleware for the running instance of Hoverfly.
```
Comment (tjcunliffe) [inline docs/reference/api.md]: Maybe rephrase to say "Gets the middleware executable and file path..." (i.e. not the middleware itself)

Comment (benjih) [inline docs/reference/api.md]: The latest commit has expanded this part of the documentation to explain what exactly the value is for middleware.

File: docs/reference/api.md [hunk 2]
```diff
@@ -1,3 +1,87 @@
...
+```
+{
+	"middleware": "python ~/middleware.py"
+}
+```
+
+## PUT /api/v2/hoverfly/middleware
+Puts the new middleware and overwrites current middleware for the running instance of Hoverfly. This requires a JSON body on the request.
```
Comment (benjih) [inline docs/reference/api.md]: The latest commit has expanded this part of the documentation to explain what exactly the value is for middleware.
